### ISSUE TITLE
Add wrapper functions for validation helper without Diagnostics

### ISSUE THREAD

**Issue by pavel-github** [hashicorp/terraform-plugin-sdk#534]
### SDK version

```
github.com/hashicorp/terraform-plugin-sdk/v2
v2.0.0
```

### Use-cases

The field `ValidateFunc` of Schema type is deprecated since v2. Now we should use `ValidateDiagFunc`, but all validation helper (except map helper) use `schema.SchemaValidateFunc` and not `schema.SchemaValidateDiagFunc`.

### Attempted Solutions

What we tried to do in our plugin was create function that wraps existing validation helpers and returns `diag.Diagnostics` instead of errors.

### Proposal

To help with migration from v1 to v2 we could create wrapper functions for every validation helper that return `schema.SchemaValidateDiagFunc` instead of `schema.SchemaValidateFunc`.

**Comment by birchb1024** [comment]
Maybe something like this:
```
				ValidateDiagFunc : validateV1(validation.IsURLWithHTTPorHTTPS),
```

Would this sort of curry work?
```
func validateV1(fn schema.SchemaValidateFunc) (schema.SchemaValidateDiagFunc) {
	return func(v interface{}, path cty.Path) diag.Diagnostics {
		var diags diag.Diagnostics

		warnings, errors := fn(v, fmt.Sprintf("%#v", path))
		for _, w := range warnings {
			diags = append(diags, diag.Diagnostic{
				Severity:      diag.Warning,
				Summary:       w,
				Detail:        w,
				AttributePath: path,
			})
		}
		for _, err := range errors {
			diags = append(diags, diag.Diagnostic{
				Severity:      diag.Warning,
				Summary:       err.Error(),
				Detail:        err.Error(),
				AttributePath: path,
			})
		}
		return diags
	}
}

```

**Comment by risicle** [comment]
Similarly, the `All()` and `Any()` functions only take `schema.SchemaValidateFunc`, which is pretty annoying.

**Comment by paddycarver** [comment]
This was released with version 2.4.0 of the SDK.
